hivemq-edge/src/frontend/src/components/rjsf/BatchSubscription/components/DataSourceStep.tsx

```diff
@@ -11,16 +18,84 @@ const getDropZoneBorder = (color: string) => {
   }
 }
 
-const DataSourceStep: FC = () => {
+const DataSourceStep: FC<StepProps> = ({ onContinue }) => {
+  const { t } = useTranslation('components')
+  const toast = useToast()
+  const [loading, setLoading] = useState(false)
+  const { getRootProps, getInputProps, isDragActive, open } = useDropzone({
+    noClick: true,
+    noKeyboard: true,
+    maxFiles: 1,
+    accept: acceptMimeTypes,
+    onDropRejected: (fileRejections) => {
+      const status: ToastStatus = 'error'
+      setLoading(false)
+      fileRejections.forEach((fileRejection) => {
+        toast({
+          ...DEFAULT_TOAST_OPTION,
+          status,
+          title: t('rjsf.batchUpload.dropZone.status', { context: status, fileName: fileRejection.file.name }),
+          description: fileRejection.errors[0].message,
+        })
+      })
+    },
+    onDropAccepted: async (files) => {
+      const [file] = files
+      setLoading(true)
+      try {
+        const workbook = XLSX.read(await readFileAsync(file), {
+          dense: true,
+        })
+        const worksheet = workbook.Sheets[workbook.SheetNames[0]] // get the first worksheet
+        const jsonData = XLSX.utils.sheet_to_json<WorksheetData>(worksheet) // generate objects
+
+        console.log('xxxxxx', jsonData)
+        const status: ToastStatus = 'success'
+        toast({
+          ...DEFAULT_TOAST_OPTION,
+          status,
+          title: t('rjsf.batchUpload.dropZone.status', { context: status, fileName: file.name }),
+        })
+        onContinue({ worksheet: jsonData })
+      } catch (error) {
+        let message
+        if (error instanceof Error) message = error.message
+        else message = String(error)
+        const status: ToastStatus = 'error'
+        toast({
+          ...DEFAULT_TOAST_OPTION,
+          status,
+          title: t('rjsf.batchUpload.dropZone.status', { context: status, fileName: file.name }),
+          description: message,
+        })
+      }
+
+      setLoading(false)
+    },
+  })
+
   return (
     <VStack
-      color="red"
+      {...getRootProps()}
       {...getDropZoneBorder('blue.500')}
       minHeight="calc(450px - 2rem)"
       display="flex"
       justifyContent="center"
       alignItems="center"
-    ></VStack>
+      id="dropzone"
+    >
+      <input {...getInputProps()} data-testid="batch-load-dropzone" />
+      {isDragActive ? (
+        <Text>{t('rjsf.batchUpload.dropZone.dropping')}</Text>
+      ) : loading ? (
+        <Text>{t('rjsf.batchUpload.dropZone.loading')}</Text>
+      ) : (
+        <>
+          <Text>{t('rjsf.batchUpload.dropZone.placeholder')}</Text>
+          <Button onClick={open}>{t('rjsf.batchUpload.dropZone.selectFile')}</Button>
+        </>
+      )}
```

Comment:
suggestion(non-blocking): A bit hard to parse in my head, WDYT about the following?

```suggestion
      {isDragActive && (
        <Text>{t('rjsf.batchUpload.dropZone.dropping')}</Text>
      )}
      {loading && (
        <Text>{t('rjsf.batchUpload.dropZone.loading')}</Text>
      )}
      {!isDragActive && !loading && (
        <>
          <Text>{t('rjsf.batchUpload.dropZone.placeholder')}</Text>
          <Button onClick={open}>{t('rjsf.batchUpload.dropZone.selectFile')}</Button>
        </>
      )}
```